app/helpers/cms/form_helper.rb

```diff
@@ -1,59 +1,29 @@
+# frozen_string_literal: true
+
 module CMS
   module FormHelper
 
     def cms_form_for(template, options = {}, &block)
-      base_name = template.class.base_class.model_name.to_s.parameterize(separator: '_').to_sym
-
       options[:builder] ||= CMS::SemanticFormBuilder
       options[:url] ||= polymorphic_path([:provider, :admin, template])
-      options[:as] ||= base_name
+      options[:as] ||= base_name(template)
 
       semantic_form_for(template, options, &block)
     end
 
     def cms_remote_form_for(template, options = {}, &block)
-      cms_form_for(template, options.reverse_merge(:html => { :"data-remote" => 'true' }), &block)
+      cms_form_for(template, options.reverse_merge({ remote: true }), &block)
     end
 
     def cms_fields_for(template, options = {}, &block)
-      base_name = template.class.base_class.model_name.to_s.parameterize(separator: '_').to_sym
-
       options[:builder] ||= CMS::SemanticFormBuilder
-      options[:as] ||= base_name
 
-      monkey_patched_semantic_fields_for(template, options, &block)
+      semantic_fields_for(base_name(template), template, options, &block)
     end
 
-    private
-
-    # Method `fields_for` changed the number of parameters in Rails3 and therefor is not
-    # compatible with our version of Formtastic. Remove this when we upgrade
-    # Formtastic.
-    #
-    def monkey_patched_semantic_fields_for(record_or_name_or_array, *args, &proc)
```

Comment:
I believe we already upgraded Rails as well as Formtastic enough to make this monkey-patch unnecessary :upside_down_face:

Reply:
I think this might be the related fix: https://github.com/formtastic/formtastic/commit/d674c4a2db51feef3d2b9f458f9db867fa2da495

Now, most of this code is for `semantic_form_for`, not for `semantic_fields_for`, I think :thinking: See https://github.com/formtastic/formtastic/blob/4.0.0/lib/formtastic/helpers/form_helper.rb#L153-L190

But I think this is probably wrong anyway.